fe/fe-core/src/main/java/org/apache/doris/analysis/DistributionDesc.java

```diff
@@ -52,27 +70,10 @@ public DistributionInfo toDistributionInfo(List<Column> columns) throws DdlExcep
         throw new NotImplementedException();
     }
 
-    public static DistributionDesc read(DataInput in) throws IOException {
-        DistributionInfoType type = DistributionInfoType.valueOf(Text.readString(in));
-        if (type == DistributionInfoType.HASH) {
-            DistributionDesc desc = new HashDistributionDesc();
-            desc.readFields(in);
-            return desc;
-        } else if (type == DistributionInfoType.RANDOM) {
-            DistributionDesc desc = new RandomDistributionDesc();
-            desc.readFields(in);
-            return desc;
-        } else {
-            throw new IOException("Unknown distribution type: " + type);
-        }
-    }
-
     @Override
     public void write(DataOutput out) throws IOException {
         Text.writeString(out, type.name());
-    }
-
-    public void readFields(DataInput in) throws IOException {
-        throw new NotImplementedException();
+        out.writeInt(numBucket);
```

Comment:
You can not change `write` method like this, this will cause metadata incompatible.
I suggest to not modify the `write` and `read` method of `DistributionDesc` and its derived classes.
And also leave the `numBucket` and `autoBucket` in derived classes.
And you can add `getBuckets()` and `setBuckets()` method and override them in derived classes.
This maybe duplicated, but will not broke the metadata compatibility.

Reply:
I check all the code， FE not use DistributionDesc write/read. It use DistributionInfo write/read function, so I not add autoBucket in these functions, just add them in property and load it in OlapTable read with markAutoBucket